src/CMakeLists.txt

```diff
@@ -27,5 +27,6 @@ endif()
 target_link_libraries(core PRIVATE readers common)
 target_link_libraries(core PUBLIC reader-interface)
 target_link_libraries(renderer PRIVATE core common CONAN_PKG::glm CONAN_PKG::stb CONAN_PKG::glad)
-target_link_libraries(readers PRIVATE reader-interface common ${BOOST_LIBRARY_TO_LINK})
+target_link_libraries(readers PRIVATE reader-interface common ${BOOST_LIBRARY_TO_LINK} core)
```

Comment:
we dont need reader-interface at all anymore and delete it declaration in `src/core/CMakeList.txt`